webui/react/src/pages/ExperimentComparison/CompareVisualization.tsx

```diff
@@ -205,9 +205,8 @@ const CompareVisualization: React.FC = () => {
         );
         setChartData(newChartData);
       },
-    ).catch((e) => {
-      setPageError(e);
-    });
+      () => setPageError(PageError.ExperimentSample),
+    );
```

Comment:
errors weren't getting thrown by the `readStream` code.

instead we now pass a handler and call it where appropriate.